horovod/runner/mpi_run.py

```diff
@@ -110,13 +119,15 @@ def _get_mpi_implementation(env=None):
 
 def _get_mpi_implementation_flags(tcp_flag, env=None):
     if is_open_mpi(env):
-        return list(_OMPI_FLAGS), list(_NO_BINDING_ARGS)
+        return list(_OMPI_FLAGS), list(_NO_BINDING_ARGS), _OMPI_IMPL
     elif is_spectrum_mpi(env):
-        return list(_SMPI_FLAGS) if not tcp_flag else list(_SMPI_FLAGS_TCP), list(_SOCKET_BINDING_ARGS)
+        return list(_SMPI_FLAGS) if not tcp_flag else list(_SMPI_FLAGS_TCP), list(_SOCKET_BINDING_ARGS), _SMPI_IMPL
```

Comment:
brackets might be needed here:

    (list(_SMPI_FLAGS) if not tcp_flag else list(_SMPI_FLAGS_TCP)), list(_SOCKET_BINDING_ARGS), _SMPI_IMPL

Even if they are not needed semantically, they would help to clarify what is meant in this condition.

While you are at it, you could flip the condition:

    (list(_SMPI_FLAGS_TCP) if tcp_flag else list(_SMPI_FLAGS)), list(_SOCKET_BINDING_ARGS), _SMPI_IMPL

Reply:
Agree, thanks